### ISSUE TITLE
How to load model for fine tuning

### ISSUE THREAD

**Issue by dtdo90** [lyuwenyu/RT-DETR#502]
Hi. Thank you for the amazing work!

I'm learning the model and want to fine-tune it on custom dataset. However, I have trouble loading the model for training. I couldn't run forward pass without .deploy() (as I find from the code).

Very much appreciate if you could advise on this issue!

import torch
import torch.nn as nn
from your_project.config import YAMLConfig  # Replace with actual import path

class Model(nn.Module):
    def __init__(self, cfg):
        super().__init__()
        self.model = cfg.model.deploy()
        self.postprocessor = cfg.postprocessor.deploy()

    def forward(self, images, orig_target_sizes):
        outputs = self.model(images)
        outputs = self.postprocessor(outputs, orig_target_sizes)
        return outputs

# Define the arguments directly as variables
config_path = "rtdetrv2_pytorch/configs/rtdetrv2/rtdetrv2_r18vd_120e_coco.yml"
resume_path = "rtdetrv2_pytorch/models/rtdetrv2_r18vd_120e_coco_rerun_48.1.pth"

# Load the configuration
cfg = YAMLConfig(config_path, resume=resume_path)

# Load the model state_dict
checkpoint = torch.load(resume_path, map_location="cpu")
if "ema" in checkpoint:
    state = checkpoint["ema"]["module"]
else:
    state = checkpoint["model"]

# Load train mode state and convert to deploy mode
model = cfg.model.load_state_dict(state)

# Instantiate the model
model = Model(cfg)

**Comment by lyuwenyu** [comment]
see this doc. https://github.com/lyuwenyu/RT-DETR/tree/main/rtdetrv2_pytorch#usage

**Comment by dtdo90** [comment]
Thank you for replying. Sorry I wasn't sure how to post the code message.

I think I get the answer for my question. I want to load the model and run a forward pass through it without using .deploy(). I get an error in the function  get_contrastive_denoising_training_group which needs targets. In training, I will always need targets to run the model?

<img width="832" alt="Screenshot 2024-12-02 at 4 04 07 PM" src="https://github.com/user-attachments/assets/bc58351c-9e6a-46da-b96e-8f8fefe6204b">

**Comment by lyuwenyu** [comment]
Yes, in training mode, the model needs to compute the loss.

**Comment by dtdo90** [comment]
Thank you very much for the quick reply! 

I adjusted _num_classes_, _img_folder_, _ann_file_ in _coco_detection.yml_ to suit my data. Would this load an **untrained model** or the **trained model**?

<img width="840" alt="Screenshot 2024-12-05 at 5 51 15 PM" src="https://github.com/user-attachments/assets/9b30ed9f-e274-402a-bfa0-71964f2d2117">
